src/styles/julabat.js

```diff
@@ -5,94 +5,99 @@
+import { getWeightedOption } from '@liamegan1/fxhash-helpers'
 
-const HUES = [
-  [[0, 20, 35, 50, 160, 180, 220, 240, 345], 1],
-  [[220, 230, 245, 255, 350], 3],
-  [[15, 20, 25, 30, 35, 40, 45, 50, 170, 235], 3],
-  [[350, 175, 230], 3]
-]
+const noiseField = []
 
 export default class JuLabatStyle extends Style {
   constructor (gridSizeX, gridSizeY, s, projectionCalculator3d, p5) {
     super(gridSizeX, gridSizeY, s, projectionCalculator3d, p5)
-    this.hues = getWeightedOption(HUES)
-    this.noiseField = []
     this.bgColor = '#080808'
-    this.offset = 0
+    this.HUES = [
+      [[0, 20, 35, 50, 160, 180, 220, 240, 345], 1],
+      [[220, 230, 245, 255, 350], 3],
+      [[15, 20, 25, 30, 35, 40, 45, 50, 170, 235], 3],
+      [[350, 175, 230], 3]
+    ]
+    this.hues = getWeightedOption(this.HUES)
+    this.offset = this._s / 100
+    this.starsGlow = false
+    this.tileGlow = true
   }
 
   beforeDraw () {
     const P5 = this._p5
+    P5.loop()
 
     P5.colorMode(P5.HSB)
     P5.background(this.bgColor)
 
-    this.offset = P5.height / 100
-
     // Fill noiseField
-    const inc = FXRandomBetween(0.01, 0.03)
-    const def = 8
-    let yoff = 0
-    for (let j = 0; j < P5.height; j += def) {
-      let xoff = 0
-      for (let i = 0; i < P5.width; i += def) {
-        const v = P5.noise(xoff, yoff)
-        // Store slightly randomized position + noise value as vector's z
-        const point = P5.createVector(
-          i + FXRandomBetween(-def * 6, def * 6),
-          j + FXRandomBetween(-def * 6, def * 6),
-          v
-        )
-        this.noiseField.push(point)
-        xoff += inc
+    if (noiseField.length === 0) {
+      const inc = P5.random(0.01, 0.03)
+      const def = 8
+      let yoff = 0
+      for (let j = 0; j < this._s; j += def) {
+        let xoff = 0
+        for (let i = 0; i < this._s; i += def) {
+          const val = P5.noise(xoff, yoff)
+          // Store slightly randomized position + noise value as vector's z
+          const point = P5.createVector(
+            (i + P5.random(0, def * 5)) / this._s,
+            (j + P5.random(0, def * 5)) / this._s,
+            val
+          )
+
+          if (point.x > 0 && point.y > 0 && point.x < this._s && point.y < this._s) noiseField.push(point)
+          xoff += inc
+        }
+        yoff += inc
       }
-      yoff += inc
     }
 
     // Compute average noiseField noise value
-    const sum = this.noiseField.reduce((a, v) => a + v.z, 0)
-    const avg = (sum / this.noiseField.length)
+    const sum = noiseField.reduce((a, v) => a + v.z, 0)
+    const avg = sum / noiseField.length
     // Create star with random chance depending on noise value
-    const f = this.noiseField.reduce((a, v) => {
-      const r = FXRandomBetween(0, 1)
-      if (v.z < r * 0.5 && v.z < avg) a.push(v)
+    const f = noiseField.reduce((a, v) => {
+      if (v.z < P5.random() * 0.5 && v.z <= avg) a.push(v)
       return a
     }, [])
 
+    console.log(f.length)
```

Comment:
Can you remove all the useless console.log call please?

Reply:
I missed this one, removing it right now